File: solutions/c/square-root/1/square_root.c
#include "square_root.h"

#include <stdint.h>
#include <math.h>

uint16_t square_root(uint16_t n) {
    double x = 0.75*n;
    for(int i = 0; i<10; i++){
        x  = 0.5 *(x + n/x);
    }
    return floor(x);
}

File: solutions/free-pascal/square-root/1/SquareRoot.pas
unit SquareRoot;

{$mode ObjFPC}{$H+}

interface

function SquareRoot(const n : integer) : integer;

implementation

uses SysUtils, Math;

function SquareRoot(const n : integer) : integer;
var 
  x: double;
begin
  x := 0.75*n;
  while floor(x)*floor(x)<>n do
  begin
    x:= 0.5*(x + n/x);
  end;
  result:= floor(x);
end;

end.
